### PR TITLE
F#src 3193 merge mocks

### PR DESCRIPTION
**Pull request by carebare47**

### CODE REVIEW

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import geometry_msgs.msg
+import termios
+from std_srvs.srv import Trigger, TriggerRequest
+from threading import Thread, Lock
+from sr_vive_tracking.msg import ControllerEvent
+from control_msgs.msg import FollowJointTrajectoryAction, FollowJointTrajectoryGoal
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+
+
+class VRCodes:
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: this is not used

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        controller_event = ControllerEvent()
+        controller_event.header.stamp = rospy.Time.now()
+        controller_event.device_id = device_id
+        controller_event.button_id = button_id
+        controller_event.event_id = event_id
+        controller_event.device_type = device_type
+        controller_event.device_index = device_index
+        self._button_publisher.publish(controller_event)
+
+    def trigger_press(self):
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: neither are those :)

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        self._publish_button_event(0, VRCodes.k_EButton_SteamVR_Trigger, VRCodes.VREvent_ButtonPress,
+                                   VRCodes.TrackedDeviceClass_Controller, 0)
+    def trigger_release(self):
+        self._publish_button_event(0, VRCodes.k_EButton_SteamVR_Trigger, VRCodes.VREvent_ButtonUnpress,
+                                   VRCodes.TrackedDeviceClass_Controller, 0)
+
+    def run(self):
+        if self._cycle_hand_positions:
+            while not rospy.is_shutdown():
+                #self.trigger_press()
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: please remove commended out code

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                self.pack_hand()
+                rospy.loginfo("SPAM")
+                rospy.sleep(3)
+                self.open_hand()
+                rospy.loginfo("SPAM")
+                rospy.sleep(3)
+
+            #self.trigger_release()
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: and here

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    def trigger_release(self):
+        self._publish_button_event(0, VRCodes.k_EButton_SteamVR_Trigger, VRCodes.VREvent_ButtonUnpress,
+                                   VRCodes.TrackedDeviceClass_Controller, 0)
+
+    def run(self):
+        if self._cycle_hand_positions:
+            while not rospy.is_shutdown():
+                #self.trigger_press()
+                self.pack_hand()
+                rospy.loginfo("SPAM")
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: remove spam :P

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                                       "rh_THJ2", "rh_THJ3", "rh_THJ4", "rh_THJ5", "rh_WRJ1", "rh_WRJ2"]
+        joint_traj_point = JointTrajectoryPoint()
+
+        joint_traj_point.positions = hand_positions
+
+        joint_traj_point.time_from_start = rospy.Duration(0.1)
+        goal.trajectory.points.append(joint_traj_point)
+        self._hand_traj_client.send_goal(goal)
+
+    def _publish_button_event(self, device_id, button_id, event_id, device_type, device_index):
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: not used

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    VREvent_ButtonPress = 200
+    VREvent_ButtonUnpress = 201
+    TrackedDeviceClass_Controller = 2
+
+class CybergloveMock(object):
+    def __init__(self, cycle_hand_positions=True):
+        self._cycle_hand_positions = cycle_hand_positions
+
+        self._button_publisher = rospy.Publisher('sr_vive_controller_events', ControllerEvent, queue_size=5)
+        self._hand_traj_client = actionlib.SimpleActionClient('/sr_teleop_vive_cyberglove_trajectory_controller' +
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: this should really be `"rh_trajectory_controller/follow_joint_trajectory"` by default. Only when running it for teleop, we should remap that to this value, like is being done here https://github.com/shadow-robot/sr_teleop_demos/blob/kinetic-devel/sr_teleop_vive_cyberglove/launch/cyberglove.launch#L9

Guess you should just parametrize it and in the top level mock for teleop overwrite to proper value.

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 7 above)
Comment (mikramarc) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: Also, you will have to change that default here:
https://github.com/shadow-robot/sr_teleop_demos/pull/47/files#diff-548d6b3dbc683e6303cde66c0ff034f7R19

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 1 above)
Comment (carebare47) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: done

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 2 above)
Comment (carebare47) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: done

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 3 above)
Comment (carebare47) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: done

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 4 above)
Comment (carebare47) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: done

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 5 above)
Comment (carebare47) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: Really? :(

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py (hunk 6 above)
Comment (carebare47) [reply cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: done

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2019 Shadow Robot Company Ltd - All Rights Reserved.
+# Proprietary and Confidential. Unauthorized copying of the content in this file, via any medium is strictly prohibited.
+
+import rospy
+import rospkg
+import argparse
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: unused

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 9]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2019 Shadow Robot Company Ltd - All Rights Reserved.
+# Proprietary and Confidential. Unauthorized copying of the content in this file, via any medium is strictly prohibited.
+
+import rospy
+import rospkg
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: unused

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+import argparse
+import actionlib
+import sys
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: unused

**Review by mikramarc — CHANGES_REQUESTED** [review]

**Review by mikramarc — APPROVED** [review]

File: cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+# Proprietary and Confidential. Unauthorized copying of the content in this file, via any medium is strictly prohibited.
+
+import rospy
+import actionlib
+from control_msgs.msg import FollowJointTrajectoryAction, FollowJointTrajectoryGoal
+from trajectory_msgs.msg import JointTrajectoryPoint
+
+class CybergloveMock(object):
+    def __init__(self, cycle_hand_positions=True):
+        self._cycle_hand_positions = cycle_hand_positions
```
Comment (mikramarc) [inline cyberglove_trajectory/src/cyberglove_trajectory/cyberglove_mock.py]: If this is set to false, the node will do nothing at the moment. Should we just remove that param?

**Review by mikramarc — CHANGES_REQUESTED** [review]

**Review by mikramarc — APPROVED** [review]

**Review by mikramarc — APPROVED** [review]

**Pull request merged**